onnxruntime/contrib_ops/rocm/bert/fast_gelu_tunable_op.h

```diff
@@ -36,11 +39,12 @@ template <typename T, int ThreadsPerBlock, int VecSize>
 Status FastGeluOp(const FastGeluParams<T>* params) {
   // TODO(anyone): Add tail handling for FastGelu
   TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
-      !((params->bias_length > 0 && params->bias_length % VecSize == 0 && params->input_length % VecSize == 0) ||
-        (params->bias_length == 0 && params->input_length % VecSize == 0)));
+      !(((params->bias_length > 0 && params->bias_length % VecSize == 0 && params->input_length % VecSize == 0) ||
+         (params->bias_length == 0 && params->input_length % VecSize == 0)) &&
+        params->input_length > (ThreadsPerBlock - GPU_WARP_SIZE) * VecSize));
```

Comment:
Can you split this into multiple `TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF`?

Or if possible, you should opt for non-compact `IsSupported` Impl and use a functor for this purpose.

Reply:
Okay, I think `IsSupported ` impl is better.